From e1314a0cef56f35b08bc69b9580f617af5940f60 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:38:23 -0400
Subject: [virt] kvm/x86: trace cpuid emulation when called from emulator

Message-id: <1406040016-3289-109-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86187
O-Subject: [RHEL7 PATCH v2 108/221] KVM: x86: trace cpuid emulation when called from emulator
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Gleb Natapov <gleb@redhat.com>

Currently cpuid emulation is traced only when executed by intercept.
Move trace point so that emulator invocation is traced too.

Reviewed-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Gleb Natapov <gleb@redhat.com>
(cherry picked from commit a9d4e4393b13bca899ab522aa812138437a6c188)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/cpuid.c b/arch/x86/kvm/cpuid.c
index cf21712..450f70a 100644
--- a/arch/x86/kvm/cpuid.c
+++ b/arch/x86/kvm/cpuid.c
@@ -762,6 +762,7 @@ void kvm_cpuid(struct kvm_vcpu *vcpu, u32 *eax, u32 *ebx, u32 *ecx, u32 *edx)
   *edx = best->edx;
  } else
   *eax = *ebx = *ecx = *edx = 0;
+ trace_kvm_cpuid(function, *eax, *ebx, *ecx, *edx);
 }
 EXPORT_SYMBOL_GPL(kvm_cpuid);
 
@@ -777,6 +778,5 @@ void kvm_emulate_cpuid(struct kvm_vcpu *vcpu)
  kvm_register_write(vcpu, VCPU_REGS_RCX, ecx);
  kvm_register_write(vcpu, VCPU_REGS_RDX, edx);
  kvm_x86_ops->skip_emulated_instruction(vcpu);
- trace_kvm_cpuid(function, eax, ebx, ecx, edx);
 }
 EXPORT_SYMBOL_GPL(kvm_emulate_cpuid);
-- 
1.7.1